qiskit/result/result.py

```diff
@@ -62,8 +62,7 @@ def __init__(
         self.job_id = job_id
         self.success = success
         self.results = results
-        if date is not None:
-            self.date = date
+        self.date = date
         if status is not None:
             self.status = status
         if header is not None:
```

Comment:
Please can we do this for `status` and `header` as well, so we're consistent across the whole class?